src/form/__tests__/analytics.test.tsx

```diff
@@ -0,0 +1,144 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React from 'react';
+import { render } from '@testing-library/react';
+
+import createWrapper from '../../../lib/components/test-utils/dom';
+import Button from '../../../lib/components/button';
+import Form from '../../../lib/components/form';
+
+import { FunnelMetrics, setFunnelMetrics } from '../../../lib/components/internal/analytics';
+
+const mockedFunnelInteractionId = 'mocked-funnel-id';
+function mockFunnelMetrics() {
+  setFunnelMetrics({
+    funnelStart: jest.fn(() => mockedFunnelInteractionId),
+    funnelError: jest.fn(),
+    funnelComplete: jest.fn(),
+    funnelSuccessful: jest.fn(),
+    funnelCancelled: jest.fn(),
+    funnelStepStart: jest.fn(),
+    funnelStepComplete: jest.fn(),
+    funnelStepNavigation: jest.fn(),
+    funnelSubStepStart: jest.fn(),
+    funnelSubStepComplete: jest.fn(),
+    funnelSubStepError: jest.fn(),
+    helpPanelInteracted: jest.fn(),
+    externalLinkInteracted: jest.fn(),
+  });
+}
+
+describe('Form Analytics', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    mockFunnelMetrics();
+  });
+
+  test('sends funnelStart and funnelStepStart metrics when Form is mounted', () => {
+    render(<Form />);
+
+    expect(FunnelMetrics.funnelStart).toHaveBeenCalledTimes(1);
+    expect(FunnelMetrics.funnelStart).toHaveBeenCalledWith(
+      expect.objectContaining({
+        funnelType: 'single-page',
+        totalFunnelSteps: 1,
+        optionalStepNumbers: [],
+        funnelNameSelector: expect.any(String),
+        funnelVersion: expect.any(String),
+        componentVersion: expect.any(String),
+        componentTheme: expect.any(String),
+      })
+    );
+
+    expect(FunnelMetrics.funnelStepStart).toHaveBeenCalledTimes(1);
+    expect(FunnelMetrics.funnelStepStart).toHaveBeenCalledWith(
+      expect.objectContaining({
+        stepNumber: 1,
+        funnelInteractionId: expect.any(String),
+        stepNameSelector: expect.any(String),
+        subStepAllSelector: expect.any(String),
+      })
+    );
+  });
+
+  test('sends a funnelStepComplete metric when Form is unmounted', () => {
+    const { unmount } = render(<Form />);
+
+    unmount();
+
+    expect(FunnelMetrics.funnelStepComplete).toHaveBeenCalledTimes(1);
+    expect(FunnelMetrics.funnelStepComplete).toHaveBeenCalledWith(
+      expect.objectContaining({
+        stepNumber: 1,
+        stepNameSelector: expect.any(String),
+        subStepAllSelector: expect.any(String),
+        funnelInteractionId: expect.any(String),
+      })
+    );
+  });
+
+  test('sends a funnelCancelled metric when Form is unmounted without clicking submit', () => {
+    const { unmount } = render(<Form />);
+
+    // funnelCancel is called in the next tick after unmounting
+    unmount();
+
```

Comment:
It is like this because a Form is both the funnel and a single step so I could move the assertions into a single test but its more that funnelStepComplete is always expected to fire when the form unmounts but funnelCancelled is only expected to fire if the Submit button is not clicked.